### PR TITLE
Kit Provider added, removes v1 support, 

### LINKED ISSUE #339: `checking cloud config` output checks the same thing multiple times
```
[my-environment] running cloud-config and environmental parameters checks...
--
  [Checking cloud config]
--
  ✔  vm_type 'small' exists
--
  ✔  vm_type 'small' exists
--
  ✔  vm_type 'small' exists
--
  ✔  vm_type 'small' exists
--
  ✔  vm_type 'small' exists
--
  ✔  vm_type 'small' exists
--
  ✔  vm_type 'small' exists
--
  ✔  vm_type 'medium' exists
--
  ✔  vm_type 'medium' exists
--
  ✔  vm_type 'medium' exists
--
  ✔  vm_type 'medium' exists
--
  ✔  vm_type 'medium' exists
--
  ✔  disk_type 'blobstore' exists
--
  ✔  vm_type 'large' exists
--
  ✔  disk_type 'postgres' exists
--
  ✔  network 'cf-db' exists
--
  ✔  vm_type 'runtime' exists
--
  ✔  network 'cf-core' exists
--
  ✔  network 'cf-edge' exists
--
  ✔  network 'cf-runtime' exists
```

Only checking each type once is both a performance optimization and makes this output more understandable.

### PR DESCRIPTION
**Pull request by dennisjbell**
# New Feature: Kit Providers

Until now, you could only use the kits that were provided by the
`genesis_community` organization on GitHub, or by creating dev kits yourself.
This release allows you to specify any GitHub org as a kit provider.

## Creating a new repo with an alternative kit provider:

```
genesis init --kit-provider        github \
             --kit-provider-org    myproject \
             --kit-provider-domain github.mycorp.com \
             --kit-provider-tls    skip\
             --kit-provider-name   "Our Custom Kit Provider" \
             --kit                 my-awesome-kit
```

There are further options, which can be shown using --help.

## Showing current kit provider:

```
genesis kit-provider

Collecting information on current kit provider for concourse deployment at ./work/concourse-deployments ...done.

         Type: github

         Name: Our Custom Kit Provider
       Domain: github.mycorp.com
          Org: myproject
      Use TLS: skip

       Status: ok

         Kits: my-awesome-kit
               another-kit

```

## Changing kit provider on existing repo:

```
genesis kit-provider --default
genesis kit-provider --kit-provider genesis-community
```

Either of these will set the kit provider to the default Genesis Community kit
provider.

```
genesis kit-provider --kit-provider github [... rest of github-specific options]
```

You can also change it to another github org as its kit provider using the
above.

Note:  You can't just change one option to have it update just that property
-- you must set all the provider-specific options to update the provider
information.  Refer to the help output (-h|--help) for details on what options
are expected.

# Removed Feature: Genesis v1 Support

This is a breaking change for anyone who still uses Genesis v1 repositories.

When Genesis v2 came out, it included full backwards compatibility with
Genesis v1 repositories, which were very different in structure and behaviour.
As v1 was a Bash script, and v2 was built in Perl, this was accomplished by
simply embedding the Bash script in the Perl (as data up to v2.4.x, and as a
separate file after that) and passing control to that script when a v1 repo
was detected.

However, with BOSH no longer supporting v1-style manifests, it seemed a good
time to remove this obsolete functionality and shrink down the package.  If you
still need this, any version from v2.5.x or v2.6.x will work for you, as the
v1 embedded in those versions haven't changed.

# Improvements

- cloud_config_needs will only check for a specific resource once, regardless
  of how many times the kit request a check for it. - Fixes #339

- Instead of not allowing Genesis to run if you have a bad $LANG env var set,
  Genesis will warn you that you may experience garbled output on non-ascii
  characters.

# Bug Fixes

- No longer allows mixed usage of `kit.features` and deprecated `kit.subkits`.
  Occurrences of this were rare, but caused anything defined under subkits to
  be ignored without warning.  This now causes an error.

### CODE REVIEW

File: lib/Genesis.pm [hunk 1]
```diff
@@ -561,6 +562,15 @@ sub chmod_or_fail {
 		or die "Could not change mode of $path: $!\n";
 }
 
+sub humanize_path {
```
Comment (dennisjbell) [inline lib/Genesis.pm]: Add test for at least the regression that was fixed.

File: ci/release_notes.md [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+genesis init --kit-provider        github \
+             --kit-provider-org    myproject \
+             --kit-provider-domain github.mycorp.com \
+             --kit-provider-tls    skip\
+             --kit-provider-name   "Our Custom Kit Provider" \
+             --kit                 my-awesome-kit
+```
+
+There are further options, which can be shown using --help.
+
```
Comment (jhunt) [inline ci/release_notes.md]: Can we document kit providers in detail on genesisproject.io, and link there from here?

File: lib/Genesis/Kit/Provider/GenesisCommunity.pm [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+=head1 DESCRIPTION
+
+This class represents a compiled kit, and its distribution as a tarball
+archive.  Most operators use kits in this format.
+
+=head1 CONSTRUCTORS
+
+=head2 new(%opts)
+
+Instantiates a new Kit::Compiled object.
```
Comment (jhunt) [inline lib/Genesis/Kit/Provider/GenesisCommunity.pm]: This POD looks incorrect...

File: lib/Genesis/Kit/Provider/Github.pm [hunk 4]
```diff
@@ -0,0 +1,434 @@
...
+
+# }}}
+# opts_help - specifies the new/update options understood by this provider {{{
+sub opts_help {
+	<<EOF
+  Kit Provider `github`:
+
+    This is a generic kit provider type for kits backed by Github or Github
+    Enterprise, allowing you to specify the domain url and organization to
+    target your specific provider location.  It supports the following options:
```
Comment (jhunt) [inline lib/Genesis/Kit/Provider/Github.pm]: Maybe note that this only spans a single organization.  I got the impression from our in-review conversation on Friday that I could specify multiple orgs in a single provider (which justified the pomp and circumstance around all this future flexibility 😁 )

File: lib/Genesis/Kit/Provider/GenesisCommunity.pm (hunk 3 above)
Comment (jhunt) [reply lib/Genesis/Kit/Provider/GenesisCommunity.pm]: This POD still looks incorrect...

**Comment by dennisjbell** [comment]
Withdrawing this PR, and separating out the disparate commits as individual PRs

**Pull request closed without merging**